src/Compilers/CSharp/Portable/Symbols/TypeWithAnnotations.cs

```diff
@@ -919,6 +938,103 @@ internal override void TryForceResolve(bool asValueType)
+            internal override bool IsStatic(TypeSymbol typeSymbol) => typeSymbol.IsStatic;
+            internal override bool IsVoid(TypeSymbol typeSymbol) => typeSymbol.IsVoidType();
+            internal override bool IsSZArray(TypeSymbol typeSymbol) => typeSymbol.IsSZArray();
+            internal override bool IsRefLikeType(TypeSymbol typeSymbol) => typeSymbol.IsRefLikeType;
+
+            internal NullableAnnotation GetResolvedAnnotationCore()
+            {
+                if (_typeParameter.IsNotNullable == true)
+                {
+                    return NullableAnnotation.NotAnnotated;
+                }
+                else
+                {
+                    return NullableAnnotation.Oblivious;
+                }
+            }
+
+            internal override NullableAnnotation GetResolvedAnnotation(NullableAnnotation defaultAnnotation)
+            {
+                Debug.Assert(defaultAnnotation == NullableAnnotation.Ignored);
+                if (_resolved == (int)NullableAnnotation.Ignored)
+                {
+                    Interlocked.CompareExchange(ref _resolved, value: (int)GetResolvedAnnotationCore(), comparand: (int)NullableAnnotation.Ignored);
+                }
+
+                Debug.Assert(_resolved != (int)NullableAnnotation.Ignored);
+                return (NullableAnnotation)_resolved;
+            }
+
+            internal override TypeSymbol GetNullableUnderlyingTypeOrSelf(TypeSymbol typeSymbol) => typeSymbol.StrippedType();
+            internal override TypeSymbol AsTypeSymbolOnly(TypeSymbol typeSymbol) => typeSymbol;
+            internal override bool IsResolved => _resolved != (int)NullableAnnotation.Ignored;
+            internal override TypeSymbol GetResolvedType(TypeSymbol defaultType) => defaultType;
+            internal override ImmutableArray<CustomModifier> CustomModifiers => _customModifiers;
+
+            internal override TypeWithAnnotations WithModifiers(TypeWithAnnotations type, ImmutableArray<CustomModifier> customModifiers)
+            {
+                return CreateNonLazyType(type.DefaultType, type.NullableAnnotation, customModifiers);
+            }
+
+            internal override TypeWithAnnotations WithTypeAndModifiers(TypeWithAnnotations type, TypeSymbol typeSymbol, ImmutableArray<CustomModifier> customModifiers)
+            {
+                return CreateNonLazyType(typeSymbol, type.NullableAnnotation, customModifiers);
+            }
+
+            internal override TypeWithAnnotations AsNullableReferenceType(TypeWithAnnotations type)
+            {
+                return CreateNonLazyType(type.DefaultType, NullableAnnotation.Annotated, _customModifiers);
+            }
+
+            internal override TypeWithAnnotations AsNotNullableReferenceType(TypeWithAnnotations type)
+            {
+                var defaultType = type.DefaultType;
```

Comment:
Some of these impls are copied straight from the NonLazyType. I guess we could extract them into "Core"-suffixed methods if we wanted.